x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/TransportGetProfilingAction.java

```diff
@@ -0,0 +1,343 @@
+import java.util.Set;
+
+public class TransportGetProfilingAction extends HandledTransportAction<GetProfilingRequest, GetProfilingResponse> {
+    private final NodeClient nodeClient;
+    private final ThreadContext threadContext;
+
+    @Inject
+    public TransportGetProfilingAction(TransportService transportService, ActionFilters actionFilters, NodeClient nodeClient) {
+        super(GetProfilingAction.NAME, transportService, actionFilters, GetProfilingRequest::new);
+        this.nodeClient = nodeClient;
+        this.threadContext = transportService.getThreadPool().getThreadContext();
+    }
+
+    @Override
+    protected void doExecute(Task submitTask, GetProfilingRequest request, ActionListener<GetProfilingResponse> submitListener) {
+        try (var ignored = threadContext.newTraceContext()) {
+            EventsIndex mediumDownsampled = EventsIndex.MEDIUM_DOWNSAMPLED;
+            this.nodeClient.prepareSearch(mediumDownsampled.getName())
+                .setSize(0)
+                .setQuery(request.getQuery())
+                .setTrackTotalHits(true)
+                .execute(new ActionListener<>() {
+                    @Override
+                    public void onResponse(SearchResponse searchResponse) {
+                        long sampleCount = searchResponse.getHits().getTotalHits().value;
+                        EventsIndex resampledIndex = mediumDownsampled.getResampledIndex(request.getSampleSize(), sampleCount);
+                        searchEventGroupByStackTrace(request, resampledIndex, submitListener);
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        submitListener.onFailure(e);
+                    }
+                });
+        }
+    }
+
+    private void searchEventGroupByStackTrace(
+        GetProfilingRequest request,
+        EventsIndex eventsIndex,
+        ActionListener<GetProfilingResponse> submitListener
+    ) {
+        GetProfilingResponseBuilder responseBuilder = new GetProfilingResponseBuilder();
+        this.nodeClient.prepareSearch(eventsIndex.getName())
+            .setTrackTotalHits(false)
+            .setQuery(request.getQuery())
+            .addAggregation(
+                new TermsAggregationBuilder("group_by")
+                    // 'size' should be max 100k, but might be slightly more. Better be on the safe side.
+                    .size(150_000)
+                    .field("Stacktrace.id")
+                    // 'execution_hint: map' skips the slow building of ordinals that we don't need.
+                    // Especially with high cardinality fields, this makes aggregations really slow.
+                    .executionHint("map")
+                    .subAggregation(new SumAggregationBuilder("count").field("Stacktrace.count"))
+            )
+            .addAggregation(new SumAggregationBuilder("total_count").field("Stacktrace.count"))
+            .setPreFilterShardSize(1)
```

Comment:
What is the rationale for always enabling the can match phase?

Reply:
Good catch! I took this verbatim from the Kibana plugin (see https://github.com/elastic/kibana/blame/main/x-pack/plugins/profiling/server/routes/stacktrace.ts#L138). I followed up with the team to understand its history and learned that this was introduced when we had used `composite` aggs and it (accidentally?) sped up query execution. So I believe this is obsolete here and will remove this for now.